conda/models/match_spec.py

```diff
@@ -404,6 +406,36 @@ def fn(self):
         assert val
         return val
 
+    @classmethod
+    def merge(cls, match_specs):
+        match_specs = tuple(cls(s) for s in match_specs)
+        grouped = groupby(lambda spec: spec.get_exact_value('name'), match_specs)
```

Comment:
Didn't check myself, but is the `match_specs` input always sorted by `spec.get_exact_value('name')`? `groupby` groups by consecutive keys only.

Reply:
Ah, disregard that comment, you used `(cy)toolz.itertoolz.groupby` not `itertools.groupby`, so everything is fine!